sc/social/like/subscribers.py

```diff
@@ -97,9 +98,7 @@ def assign_canonical_url(obj, event):
 
     # we can't assign a canonical URL without a canonical domain
     if canonical_domain:
-        # FIXME: we're currently ignoring the Plone site id
-        #        https://github.com/collective/sc.social.like/issues/119
-        path = '/'.join(obj.getPhysicalPath()[2:])
+        path = get_path_to_virtual_path(event.request, obj)
```

Comment:
just do the conversion right here; you'll need to get the `request` on a different way (probably using `getRequest()` fro `zope.globalrequest`) as the `AfterTransitionEvent` doesn't includes it.